core/src/test/scala/org/locationtech/rasterframes/RasterFramesStatsSpec.scala

```diff
@@ -21,46 +21,109 @@
+        .tileStat()
+        .approxTileQuantile(
+          "tile",
+          Array(0.10, 0.50, 0.90),
+          0.00001
+        )
 
-      result.length should be (3)
+      result.length should be(3)
 
       // computing externally with numpy we arrive at 7963, 10068, 12160 for these quantiles
-      result should contain inOrderOnly (7963.0, 10068.0, 12160.0)
+      result should contain inOrderOnly(7963.0, 10068.0, 12160.0)
     }
 
-    it("should compute approx percentiles for many tile cols"){
-      val result = df.tileStat().approxTileQuantile(
-        Array("tile", "tilePlus2"),
-        Array(0.25, 0.75),
-        0.00001
-      )
-      result.length should be (2)
+    it("should compute approx percentiles for many tile cols") {
+      val result = df
+        .tileStat()
+        .approxTileQuantile(
+          Array("tile", "tilePlus2"),
+          Array(0.25, 0.75),
+          0.00001
+        )
+      result.length should be(2)
       // nested inside is another array of length 2 for each p
-      result.foreach{c ⇒ c.length should be (2)}
+      result.foreach { c =>
+        c.length should be(2)
+      }
+
+      result.head should contain inOrderOnly(8701, 11261)
+      result.tail.head should contain inOrderOnly(8703, 11263)
+    }
+  }
+
+  describe("Tile quantiles through built-in functions") {
+
+    it("should compute approx percentiles for a single tile col") {
+      // Use "explode"
+      val result = df
+        .select(rf_explode_tiles($"tile"))
+        .stat
+        .approxQuantile("tile", Array(0.10, 0.50, 0.90), 0.00001)
+
+      result.length should be(3)
+
+      // computing externally with numpy we arrive at 7963, 10068, 12160 for these quantiles
+      result should contain inOrderOnly(7963.0, 10068.0, 12160.0)
+
+      // Use "to_array" and built-in explode
+      val result2 = df
+        .select(explode(rf_tile_to_array_double($"tile")) as "tile")
+        .stat
+        .approxQuantile("tile", Array(0.10, 0.50, 0.90), 0.00001)
+
+      result2.length should be(3)
+
+      // computing externally with numpy we arrive at 7963, 10068, 12160 for these quantiles
+      result2 should contain inOrderOnly(7963.0, 10068.0, 12160.0)
 
-      result.head should contain inOrderOnly (8701, 11261)
-      result.tail.head should contain inOrderOnly (8703, 11263)
     }
+  }
+
+  describe("Tile quantiles through existing RF functions") {
+    it("should compute approx percentiles for a single tile col") {
 
+      // As the number of buckets goes up, the closer we get to the "right" answer.
+      val result = df
+        .select(rf_agg_approx_histogram($"tile", 500))
+        .map(h => h.percentileBreaks(Seq(0.1, 0.5, 0.9)))
+        .first()
+
+      result.length should be(3)
+      println(result)
+      // computing externally with numpy we arrive at 7963, 10068, 12160 for these quantiles
+      // This will fail as the histogram algorithm approximates things differently (probably not as well)
```

Comment:
I would not expect this to perform well. If you look at the geotrellis.raster.histogram.StreamingHistogram.percentileBreaks they note that it is for use in generating breaks for the histogram. So it is more rough approximation by nature I would expect. I am also not sure how this would be have at the extremes of the range due to its intended use for histogram breaks. 

In any case the method is based on linear interpolation of the histogram, which I suspect may be poor in general unless the number of bins is very large.

Reply:
👍 The exercise was good to weed out that approach.